src/main/java/org/isf/ward/model/Ward.java

```diff
@@ -314,7 +318,26 @@ public int hashCode() {
 		}
 		return this.hashCode;
 	}
-	
-	
 
+	public static class WardCodeComparator implements Comparator<Ward> {
+
+		@Override
+		public int compare(Ward object1, Ward object2) {
+			if (object2.getCode() == null) {
+				return -1;
+			}
+			return object1.getCode().compareTo(object2.getCode());
```

Comment:
```suggestion
			return object2.getCode().compareTo(object1.getCode());
```
Because what if object1.getCode() returns null?

Reply:
Actually the way I use it in the GUI (`wardsList.sort(new Ward.WardDescriptionComparator())`) doesn't require the null check, but if we want to check, then we should check both objects for null, wdyt?

The order in the `compareTo`, instead, will affect the result, that will be in alphabetical order or reversed, so I would not change that.

Reply:
Or we can just use `StringUtils.compare(object1.getCode(), object2.getCode())` (from Apache Commons) that is null safe? (cfr https://www.baeldung.com/java-compare-strings#comparison-apache)

Reply:
That anyway internally does:

```
    public static int compare(final String str1, final String str2, final boolean nullIsLess) {
        if (str1 == str2) { // NOSONARLINT this intentionally uses == to allow for both null
            return 0;
        }
        if (str1 == null) {
            return nullIsLess ? -1 : 1;
        }
        if (str2 == null) {
            return nullIsLess ? 1 : - 1;
        }
        return str1.compareTo(str2);
    }
```

Reply:
Check the new version please